src/runtime/HalideBuffer.h

```diff
@@ -1754,21 +1754,36 @@ class Buffer {
     static Buffer<T, D> make_with_shape_of(Buffer<T2, D2> src,
                                            void *(*allocate_fn)(size_t) = nullptr,
                                            void (*deallocate_fn)(void *) = nullptr) {
+
+        const halide_type_t dst_type = T_is_void
+            ? src.type()
+            : halide_type_of<typename std::remove_cv<not_void_T>::type>();
+        return Buffer<>::make_with_shape_of_helper(dst_type, src.
+                                                   dimensions(), src.buf.dim,
+                                                   allocate_fn, deallocate_fn);
+    }
+
+private:
+
+    static Buffer<> make_with_shape_of_helper(halide_type_t dst_type,
```

Comment:
This is great in that it reduces the expansion explosion that happens here due to the both-src-and-dst-template-types. But we could reduce it even further if we made (say) `struct BufferHelper` that is outside `struct Buffer<>` in which we placed static-helper methods like this, so it wouldn't vary by any Buffer<> type instantiation. So in theory that would guarantee that helper methods like this got only one instantiation. (Might be overthinking it.)

Reply:
If we only ever call this static method on `Buffer<>`, which we do, then only one version of it is ever instantiated.

Reply:
Ah, I did not notice that detail. (Worth commenting on?)